avocado/core/mux.py

```diff
@@ -29,6 +36,152 @@
 REMOVE_VALUE = 1
 
 
+class MuxTree(object):
+
+    """
+    Object representing part of the tree from the root to leaves or another
+    multiplex domain. Recursively it creates multiplexed variants of the full
+    tree.
+    """
+
+    def __init__(self, root):
+        """
+        :param root: Root of this tree slice
+        """
+        self.pools = []
```

Comment:
Yes, I agree the change is atomic, but the documentation of the reason of the changes is a very very important thing.  And there are two ways to tackle that:  write a short commit message in a separate commit with the referred change (because it's then obvious), or give more context about the change in a larger commit, which I find more difficult to both authors and reviewers.

So, to summarize the discussion with stuff we already know:

1) describing the reason of the changes (if not obvious) is a *requirement* for commit messages
2) there's no "perfect size" when bundling changes and commits, so the use of one's *best* judgement when it comes to how to bundle changes into commits is *pretty important*